### ISSUE TITLE
iocage throws a RuntimeError when error output is not a tty during CLI usage

### ISSUE THREAD

**Issue by rwestlund** [iocage/iocage#277]
Using HEAD (1f72cf04dd584ab8eaeba93576b5784b4f868687):

When redirecting `stdout`, errors become a but more severe...
```
root@r2d2 /u/h/r/a/s/jails# iocage start nonexistent
nonexistent not found!

root@r2d2 /u/h/r/a/s/jails# iocage start nonexistent > /dev/null
Traceback (most recent call last):
  File "/usr/local/bin/iocage", line 10, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/iocage/cli/start.py", line 55, in cli
    ioc.IOCage(jail, rc=rc).start()
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/iocage.py", line 1107, in start
    uuid, path = self.__check_jail_existence__()
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/iocage.py", line 245, in __check_jail_existence__
    silent=self.silent)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ioc_common.py", line 78, in logit
    _callback({"level": level, "message": msg})
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ioc_common.py", line 61, in callback
    raise RuntimeError(log['message'])
RuntimeError: nonexistent not found!
```

This seems to happen for all errors:
```
root@r2d2 /u/h/r/a/s/jails# iocage get host_hostname nonexistent
nonexistent not found!

root@r2d2 /u/h/r/a/s/jails# iocage get host_hostname nonexistent > /dev/null
Traceback (most recent call last):
  File "/usr/local/bin/iocage", line 10, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/iocage/cli/get.py", line 95, in cli
    p = ioc.IOCage(jail, skip_jails=True).get(prop)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/iocage.py", line 820, in get
    uuid, path = self.__check_jail_existence__()
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/iocage.py", line 245, in __check_jail_existence__
    silent=self.silent)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ioc_common.py", line 78, in logit
    _callback({"level": level, "message": msg})
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ioc_common.py", line 61, in callback
    raise RuntimeError(log['message'])
RuntimeError: nonexistent not found!
```

**Comment by skarekrow** [comment]
I didn't look at this closely enough, the error you're getting is expected. It happens when no tty is present, I will raise a RuntimeError instead of a nice message.

**Comment by rwestlund** [comment]
@skarekrow  Why behave that way on purpose?  You don't need a tty to print an error.

I'm using iocage non-interactively in my management scripts a lot, and the stack traces are just noise.  For example, this makes sure a jail exists and has a valid configuration:
```sh
ensure_jail_exists() {
    iocage get host_hostname $1 > /dev/null || exit 1
}
```

But it gets a stack trace (as shown above).  Just print to `stderr` and `exit 1` like everyone else does.

**Comment by skarekrow** [comment]
@rwestlund Because `iocage` can be called from the library (which you do NOT want to exit on 1) and the cli. I'll reopen this as an enhancement to see if I can have it exit on 1 with a message on the cli if the error originates from there and not a normal usage of the library.

**Comment by skarekrow** [comment]
Should be solved with this commit: 97defc51bb309366fff5ed20f6f564d18fe1cdc4 👍

**Comment by rwestlund** [comment]
Thanks, that makes scripting it much easier 👍
